src/spork.cpp

```diff
@@ -138,18 +138,18 @@ int CSporkManager::ProcessSporkMsg(CSporkMessage& spork)
             // spork is active
             if (mapSporksActive[spork.nSporkID].nTimeSigned >= spork.nTimeSigned) {
                 // spork in memory has been signed more recently
-                LogPrint(BCLog::NET, "%s : spork %d (%s) in memory is more recent: %d >= %d\n", __func__,
+                LogPrint(BCLog::SPORKS, "%s : spork %d (%s) in memory is more recent: %d >= %d\n", __func__,
                           spork.nSporkID, sporkName,
                           mapSporksActive[spork.nSporkID].nTimeSigned, spork.nTimeSigned);
                 return 0;
             } else {
                 // update active spork
-                LogPrint(BCLog::NET, "%s : got updated spork %d (%s) with value %d (signed at %d)\n", __func__,
+                LogPrint(BCLog::SPORKS, "%s : got updated spork %d (%s) with value %d (signed at %d)\n", __func__,
                           spork.nSporkID, sporkName, spork.nValue, spork.nTimeSigned);
             }
         } else {
             // spork is not active
-            LogPrint(BCLog::NET, "%s : got new spork %d (%s) with value %d (signed at %d)\n", __func__,
+            LogPrint(BCLog::SPORKS, "%s : got new spork %d (%s) with value %d (signed at %d)\n", __func__,
```

Comment:
Maybe these last two lines should be `LogPrintf` ?
Since they signal a valid update of the spork value, might be good to always log it.
Also it is not a frequent event, and cannot be used to maliciously spam the logs (as the SPORK keys would be needed).

Reply:
yeah could do that

Reply:
The signature check is done below, peers can broadcast sporks with a random/empty signature with this change and spam the logs.

What can be done to log the spork update under the general `logprintf` is move this logging after the signature validation and the sporks map update (before the relay). We are sure there that the spork data is valid.
Or well, the other option is leave it as before under the `SPORKS` logging type.

Reply:
Mhm, true.
I'd move the `LogPrintf` after the signature validation.